azure-kusto-ingest/azure/kusto/ingest/ingestion_properties.py

```diff
@@ -65,18 +66,70 @@ class ColumnMapping:
     pre-created (it is recommended to create the mappings in advance and use ingestionMappingReference).
     To read more about mappings look here: https://docs.microsoft.com/en-us/azure/kusto/management/mappings"""
 
-    # Json Mapping consts
     PATH = "Path"
     TRANSFORMATION_METHOD = "Transform"
-    # csv Mapping consts
     ORDINAL = "Ordinal"
     CONST_VALUE = "ConstValue"
-    # Avro Mapping consts
     FIELD_NAME = "Field"
+
+    # TODO - these are undocumented - do they really exist?
     COLUMNS = "Columns"
-    # General Mapping consts
     STORAGE_DATA_TYPE = "StorageDataType"
 
+    VALID_MAPPINGS: Dict[str, List[IngestionMappingKind]] = {
+        ORDINAL: [IngestionMappingKind.CSV],
+        PATH: [
+            IngestionMappingKind.JSON,
+            IngestionMappingKind.AVRO,
+            IngestionMappingKind.APACHEAVRO,
+            IngestionMappingKind.SSTREAM,
+            IngestionMappingKind.PARQUET,
+            IngestionMappingKind.ORC,
+        ],
+        CONST_VALUE: [
+            IngestionMappingKind.CSV,
+            IngestionMappingKind.JSON,
+            IngestionMappingKind.AVRO,
+            IngestionMappingKind.APACHEAVRO,
+            IngestionMappingKind.SSTREAM,
+            IngestionMappingKind.PARQUET,
+            IngestionMappingKind.ORC,
+            IngestionMappingKind.W3CLOGFILE,
+        ],
+        FIELD_NAME: [
+            IngestionMappingKind.AVRO,
+            IngestionMappingKind.APACHEAVRO,
+            IngestionMappingKind.SSTREAM,
+            IngestionMappingKind.PARQUET,
+            IngestionMappingKind.ORC,
+            IngestionMappingKind.W3CLOGFILE,
+        ],
+        TRANSFORMATION_METHOD: [
+            IngestionMappingKind.JSON,
+            IngestionMappingKind.AVRO,
+            IngestionMappingKind.APACHEAVRO,
+            IngestionMappingKind.SSTREAM,
+            IngestionMappingKind.PARQUET,
+            IngestionMappingKind.ORC,
+            IngestionMappingKind.W3CLOGFILE,
+        ],
+        COLUMNS: [IngestionMappingKind.AVRO, IngestionMappingKind.APACHEAVRO],
+        STORAGE_DATA_TYPE: [
+            IngestionMappingKind.JSON,
+            IngestionMappingKind.AVRO,
+            IngestionMappingKind.APACHEAVRO,
+            IngestionMappingKind.SSTREAM,
+            IngestionMappingKind.PARQUET,
+            IngestionMappingKind.ORC,
+            IngestionMappingKind.W3CLOGFILE,
+        ],
+    }
+
+    NEEDED_PROPERTIES = invert_dict_of_lists(VALID_MAPPINGS)
+
+    CONSTANT_TRANSFORMATION_METHODS = [TransformationMethod.SOURCE_LOCATION.value, TransformationMethod.SOURCE_LINE_NUMBER.value]
```

Comment:
the name doesnt exlpain why you choose these methods? are they common to all mappings?

Reply:
These transformations replace the field entirely - meaning they don't need a `path` or a `field`, and if you use it it's wrong.

Other transformations are dependent on the `path` or `field` property, and if you use it without, it's wrong.

We actually had a similar check before the PR.

Reply:
can u add a comment for that